tests/components/rfxtrx/test_config_flow.py

```diff
@@ -1,14 +1,281 @@
 """Test the Tado config flow."""
+import os
+
+import serial.tools.list_ports
+
 from homeassistant import config_entries, data_entry_flow, setup
-from homeassistant.components.rfxtrx import DOMAIN
+from homeassistant.components.rfxtrx import DOMAIN, config_flow
 from homeassistant.helpers.device_registry import (
     async_entries_for_config_entry,
     async_get_registry,
 )
 
+from tests.async_mock import MagicMock, patch, sentinel
 from tests.common import MockConfigEntry
 
 
+def SerialConnect(self):
+    """Mock a serial connection."""
+    self.serial = True
+
+
+def SerialConnectFail(self):
+    """Mock a failed serial connection."""
+    self.serial = None
+
+
+def com_port():
+    """Mock of a serial port."""
+    port = serial.tools.list_ports_common.ListPortInfo()
+    port.serial_number = "1234"
+    port.manufacturer = "Virtual serial port"
+    port.device = "/dev/ttyUSB1234"
+    port.description = "Some serial port"
+
+    return port
+
+
+@patch(
+    "homeassistant.components.rfxtrx.rfxtrxmod.PyNetworkTransport.connect",
+    MagicMock(return_value=None),
```

Comment:
Without `MagicMock` here, I get following error

`TypeError: object MagicMock can't be used in 'await' expression`

That's why I added the `MagicMock` (which made no sense to me, because without it suggest already using `MagicMock`). Any idea what goes wrong here?

Reply:
Could it be because `PyNetworkTransport` is not async safe?

Reply:
Seems not, same error when schedule this on the executor

Reply:
I think it's because we don't accept this mock as a parameter in the test, just `hass` currently. See comment below.